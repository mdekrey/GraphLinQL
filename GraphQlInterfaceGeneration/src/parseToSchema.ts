import { parse, buildASTSchema, GraphQLSchema } from "graphql";

export function parseToSchema(graphQlSchema: string): GraphQLSchema {
  const astDocument = parse(graphQlSchema);

  const backcompatOptions = { commentDescriptions: true };
  const schema = buildASTSchema(astDocument, backcompatOptions);

  return schema;
}
