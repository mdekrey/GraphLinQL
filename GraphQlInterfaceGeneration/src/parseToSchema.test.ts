import { parseToSchema } from "./parseToSchema";
import { getStarWarsGraphQlSchemaV3, getStarWarsGraphQlSchemaV4 } from "./resources.test";

it("can parse graphql v3 schemas", async function() {
  const schema = parseToSchema(await getStarWarsGraphQlSchemaV3());
  expect(schema).not.toBeNull();
  expect(Object.keys(schema.getTypeMap())).toMatchSnapshot();
});

it("can parse graphql v4 schemas", async function() {
  const schema = parseToSchema(await getStarWarsGraphQlSchemaV4());
  expect(schema).not.toBeNull();
  expect(schema.getType("Film")!.description).toEqual("A single film.");
  expect(Object.keys(schema.getTypeMap())).toMatchSnapshot();
});
