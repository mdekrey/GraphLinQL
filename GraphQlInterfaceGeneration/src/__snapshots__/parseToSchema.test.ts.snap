// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can parse graphql v3 schemas 1`] = `
Array [
  "Query",
  "Episode",
  "Character",
  "ID",
  "String",
  "Int",
  "FriendsConnection",
  "FriendsEdge",
  "PageInfo",
  "Boolean",
  "Review",
  "SearchResult",
  "Human",
  "LengthUnit",
  "Float",
  "Starship",
  "Droid",
  "Mutation",
  "ReviewInput",
  "ColorInput",
  "Subscription",
  "__Schema",
  "__Type",
  "__TypeKind",
  "__Field",
  "__InputValue",
  "__EnumValue",
  "__Directive",
  "__DirectiveLocation",
]
`;

exports[`can parse graphql v4 schemas 1`] = `
Array [
  "Root",
  "String",
  "Int",
  "FilmsConnection",
  "PageInfo",
  "Boolean",
  "FilmsEdge",
  "Film",
  "Node",
  "ID",
  "FilmSpeciesConnection",
  "FilmSpeciesEdge",
  "Species",
  "Float",
  "Planet",
  "PlanetResidentsConnection",
  "PlanetResidentsEdge",
  "Person",
  "PersonFilmsConnection",
  "PersonFilmsEdge",
  "PersonStarshipsConnection",
  "PersonStarshipsEdge",
  "Starship",
  "StarshipPilotsConnection",
  "StarshipPilotsEdge",
  "StarshipFilmsConnection",
  "StarshipFilmsEdge",
  "PersonVehiclesConnection",
  "PersonVehiclesEdge",
  "Vehicle",
  "VehiclePilotsConnection",
  "VehiclePilotsEdge",
  "VehicleFilmsConnection",
  "VehicleFilmsEdge",
  "PlanetFilmsConnection",
  "PlanetFilmsEdge",
  "SpeciesPeopleConnection",
  "SpeciesPeopleEdge",
  "SpeciesFilmsConnection",
  "SpeciesFilmsEdge",
  "FilmStarshipsConnection",
  "FilmStarshipsEdge",
  "FilmVehiclesConnection",
  "FilmVehiclesEdge",
  "FilmCharactersConnection",
  "FilmCharactersEdge",
  "FilmPlanetsConnection",
  "FilmPlanetsEdge",
  "PeopleConnection",
  "PeopleEdge",
  "PlanetsConnection",
  "PlanetsEdge",
  "SpeciesConnection",
  "SpeciesEdge",
  "StarshipsConnection",
  "StarshipsEdge",
  "VehiclesConnection",
  "VehiclesEdge",
  "__Schema",
  "__Type",
  "__TypeKind",
  "__Field",
  "__InputValue",
  "__EnumValue",
  "__Directive",
  "__DirectiveLocation",
]
`;
